### PR TITLE
Refactored duplicate endpoints

### LINKED ISSUE #3712: Refactor duplicate endpoints.
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-orga-server

**Current behavior:**

Right now there are some duplicate endpoints that have arisen. Need to check and refactor them to avoid same endpoints refering to different views.

### PR DESCRIPTION
**Pull request by enigmaeth**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [x] The unit tests pass locally with my changes 

#### Short description of what this resolves:
Refactored duplicate endpoints mostly for Detail endpoints which were same in List

Fixes #3712.

### CODE REVIEW

File: app/api/__init__.py [hunk 1]
```diff
@@ -162,7 +159,7 @@
 
 # tax
 api.route(TaxList, 'tax_list', '/events/<int:id>/tax', '/events/<identifier>/tax')
-api.route(TaxDetail, 'tax_detail', '/tax/<int:id>', '/events/<int:event_id>/tax')
+api.route(TaxDetail, 'tax_detail', '/tax/<int:id>')
```
Comment (niranjan94) [inline app/api/__init__.py]: Why was this removed ? An event can have only one tax. Please ensure that works fine.

Comment (SaptakS) [inline app/api/__init__.py]: @enigmaeth the first link(TaxList) is supposed to be only for POST. Otherwise GET should be done from TaxDetail only.

Comment (enigmaeth) [inline app/api/__init__.py]: @niranjan94 @SaptakS 
I think that if we want to make list post only and detail get only, then we will have to change one of the endpoints name to something else like name POST list method to `events/<event_id>/taxes` just as we did with speakers-calls. Is there any other way that we can do POST only list and GET only detail with same endpoint: `events/<event_id>/tax`?
( `events/<int:id>/tax` is to be changed to `events/<int:event_id>/tax` )

Comment (poush) [inline app/api/__init__.py]: You can provide `methods = ['POST', ]` inside TaxList. This should work. Please check.

Comment (enigmaeth) [inline app/api/__init__.py]: Resolved.

**Comment by enigmaeth** [comment]
@SaptakS @niranjan94 Updated. Please check! 
Thanks.
